x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -801,6 +804,24 @@ const updateExistingDataStream = async ({
     await rolloverDataStream(dataStreamName, esClient);
   }
 
+  if (lifecycle?.data_retention) {
```

Comment:
Should we handle the scenario where the DLM is removed in a new version of a package? It's not likely to happen, but possible.

Reply:
It seems not likely to happen and we will need to add some extra permission to `kibana_system` user for that, I will prefer to keep it like this to avoid some extra complexity.